core/domain/story_domain.py

```
@@ -1238,6 +1508,85 @@ def get_ordered_nodes(self) -> List[StoryNode]:
+        Args:
+            arc_ids_order: list(str). The ordered list of arc IDs.
+
+        Raises:
+            ValueError. If an arc ID is not found.
+        """
+        old_arcs = {arc.id: arc for arc in self.arcs}
+        new_arcs = []
+        for arc_id in arc_ids_order:
+            if arc_id not in old_arcs:
+                raise ValueError(
+                    'Arc with id %s is not part of this story.' % arc_id
```

Ruff (0.15.15)

[warning] Use format specifiers instead of percent format

Replace with format specifiers

(UP031)